### DISCUSSION TITLE
Feature: unified self-registration frontend

### DISCUSSION THREAD

**Discussion by mvdkleijn** [dexidp/dex#3787] [Ideas]
I have some beef with Dex. :stuck_out_tongue: 

Unless I majorly missed something, Dex currently is not usable for anyone that needs self-registration features. This is **not** mentioned clearly enough on the site / in the docs in my opinion.

To be clear, I've used Dex in the past and love the light-weight nature and ease of deployment. :smile: 

However, I think it would make sense for **Dex to serve a self-registration page that creates the actual user in a backend using a connector**.

To be clear, I'm **not** advocating for a full user management UI or anything like that. Merely a way to fill the backend with users via a self-registration option.

From my point of view, not having a unified self-registration frontend kinda nerfs Dex for any use-case where self-registration is needed.

Any situation where there are no pre-existing users, Dex is kinda useless. That seems like a shame to me.

So concretely I'd like to ask:

- Either adjust the site and docs to clearly show the intended use-cases for Dex.
- Or (preferably) add a self-registration option using connectors to store data in a backend.

Though I can kinda predict what the Dex project's answer here will be.. :smiling_face_with_tear:

**Comment by bdube** [comment]
I've been looking through these discussions and the documentation to see what I missed. Is the login flow the only web UI that Dex has?

**Reply by mvdkleijn** [reply]
There (currently) is only a login flow in the UI. From what I remember (granted quite a while back in 2018 or so) there used to be a self-registration flow. I gathered from various issues here that the maintainers decided to scrap that some time ago.
